docs/GUIDE_FOR_KOGITO_DEVS.md

```diff
@@ -0,0 +1,352 @@
+# Introduction
+This is a list of tasks/activities that are useful or necessary 
+to know while working on the Kogito operator. 
+This document is written in an order that is recommended to 
+be worked through chronologically as a new Kogito operator developer.
+
+# Installing Kogito Dependencies
+## CRC
+CRC (CodeReady Containers) allows you to have a locally 
+running OpenShift container on your laptop which is 
+essential for testing when making changes to the operator. 
+[This article](https://developers.redhat.com/blog/2019/09/05/red-hat-openshift-4-on-your-laptop-introducing-red-hat-codeready-containers/) 
+gives you a good overview of the product as well where to 
+download it and ensure that it works. You will need to 
+[create a Red Hat account](https://www.redhat.com/wapps/ugc/register.html) 
+to download CRC.
+
+To get the command to login as an admin, you can use `crc 
+console --credentials`.
+
+## Operator
+The operator requires some dependencies in order to run. The 
+following dependencies are copied from the [operator's README](https://github.com/kiegroup/kogito-operator#requirements). 
+Please ensure that they are up-to-date and still match. Following 
+are the specific installation commands/links for Fedora:
+
+### Single Command
+Operator Courier, Go, Golint, Python 3.x, Cekit:
+```sh
+sudo dnf install go python3 python3-pip cekit
+go get -u golang.org/x/lint/golint
+pip3 install operator-courier
+```
+
+### Multiple Commands
+- [Docker](https://docs.docker.com/engine/install/fedora/) 
+- [Operator SDK](https://sdk.operatorframework.io/docs/installation/#install-from-github-release) v1.2.0
+  - make sure to set `OPERATOR_SDK_DL_URL` to v1.2.0:
+    ```sh
+    export OPERATOR_SDK_DL_URL=https://github.com/operator-framework/operator-sdk/releases/download/v1.2.0
+    ```
+- [Golangci-lint](https://golangci-lint.run/usage/install/#linux-and-windows)
+
+## Examples
+To test that your changes to the operator works, you will 
+usually want to deploy a Kogito application to OpenShift. To 
+do so, you will need to compile [a Kogito example](https://github.com/kiegroup/kogito-examples), 
+and there are some dependencies to do so. The 
+following dependencies are copied from the [process-quarkus-example's README](https://github.com/kiegroup/kogito-examples/tree/stable/process-quarkus-example#prerequisites).
+Please ensure that they are up-to-date and still match. Following 
+are the specific installation commands/links for Fedora:
+
+### Single Command
+- Maven: `sudo dnf install maven`
+
+### Multiple Commands
+- [GraalVM](https://www.graalvm.org/docs/getting-started/linux/) 
+  - set `JAVA_HOME`/`GRAALVM_HOME` environment variable
+  - [Native Image compilation](https://www.graalvm.org/reference-manual/native-image/#install-native-image) 
+    (after installing GraalVM)
+    ```sh
+    gu install native-image
+    sudo dnf install gcc glibc-devel zlib-devel libstdc++-static
+    ```
+
+# Running Operator Locally
```

Comment:
what about to mention `operator-sdk run local --watch-namespace` as well?

Reply:
What's the difference between this command, `make run` and `go run main.go`?

Reply:
> What's the difference between this command, make run and go run main.go?

Not sure, I always use `go run main.go`. That's the recommended way IIRC.

> what about to mention operator-sdk run local --watch-namespace as well?

This is not supported anymore in the current SDK version.